Cauli/Florets/Inspector/Record List/InspectorTableViewDatasource.swift

```diff
@@ -0,0 +1,80 @@
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+internal class InspectorTableViewDatasource: NSObject {
+
+    internal typealias Filter = (Record) -> (Bool)
+
+    internal private(set) var items: [Record] = [] {
+        didSet {
+            filteredItems = self.filteredItems(in: items, with: filter)
+        }
+    }
+    internal private(set) var filteredItems: [Record] = []
+    var filter: Filter? {
+        didSet {
+            filteredItems = self.filteredItems(in: items, with: filter)
+        }
+    }
+
+    private func filteredItems(in array: [Record], with filter: Filter?) -> [Record] {
+        guard let filter = filter else { return array }
+        return array.filter(filter)
+    }
+
+    internal func filter(records: [Record]) -> [Record] {
+        return filteredItems(in: records, with: filter)
+    }
+
+    internal func append(records: [Record]) {
+        items += records
+    }
+
+}
+
+extension InspectorTableViewDatasource: UITableViewDataSource {
+    internal func setup(tableView: UITableView) {
+        tableView.dataSource = self
+        let bundle = Bundle(for: InspectorTableViewController.self)
+        let nib = UINib(nibName: InspectorRecordTableViewCell.nibName, bundle: bundle)
+        tableView.register(nib, forCellReuseIdentifier: InspectorRecordTableViewCell.reuseIdentifier)
+    }
+
+    internal func record(at indexPath: IndexPath) -> Record {
+        return filteredItems[indexPath.row]
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return filteredItems.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: InspectorRecordTableViewCell.reuseIdentifier, for: indexPath) as? InspectorRecordTableViewCell else {
+            fatalError("Unable to dequeue a cell")
+        }
+        cell.record = record(at: indexPath)
```

Comment:
@brototyp I thought about highlighting the search string in the Record cells by making it bold. As we have it currently set up I don't see a way to get the search string / search terms here. Do you have an idea how to bypass this elegantly?

Reply:
That sounds like a nice Idea. What do you think about, instead of exposing a filter closure, we expose a filter-string in the Datasource and the datasource will then take care of filtering based on that String. This way we have the search string itself and all filtering (including the "how") is done in the Datasource.

Reply:
Done and also implemented highlighting.

Reply:
I like the changes. I am not super convinced by the highlighting color. I am not sure if I will be able to come up with an alternative, but I'll try and let you know once I have a result.

Reply:
What do you think about this style? We could also just just the tint color and then set the TintColor of the CauliViewController to the Cauli Purple. 

<img width="472" alt="bildschirmfoto 2019-02-16 um 16 09 20" src="https://user-images.githubusercontent.com/113456/52901555-6d362100-3205-11e9-93a8-eca0220bcf0f.png">
<img width="472" alt="bildschirmfoto 2019-02-16 um 16 07 01" src="https://user-images.githubusercontent.com/113456/52901556-6d362100-3205-11e9-8e63-4ae321a3e5ec.png">

Reply:
I like that! Had it in bold black at first but that didn't look too fancy and also caused the text to "jump". Some color accents like in your screenshots look good anyway. Do you want to push that change?

Reply:
I just pushed the change, where it will use the tintColor of the view. In a different PR we will configure Cauli in such a way, that the purple will be the tintColor of all displayed ViewController.